### PR TITLE
Added the delete dialog feature on long click of a reminder object

### PR DESCRIPTION
**Pull request by zaiddkhan**
I have added a delete functionality, a reminder now can be deleted by long clicking on it. 
- A confirmation dialog appears after long clicking and if the user confirms it by clicking confirm, the medication will be removed from the database.

### CODE REVIEW

**Comment by waseefakhtar** [comment]
Great job with this! Could you look at why the action won't work at times?

https://github.com/waseefakhtar/dose-android/assets/4093820/94d96a34-a22c-48a9-a48d-d39aa8cf24e5

**Comment by zaiddkhan** [comment]
Are you talking about the list not updating as soon as we delete the icon? I think using the collect block instead of forEach would work

**Comment by zaiddkhan** [comment]
@waseefakhtar can you accept the pull request now?

**Comment by waseefakhtar** [comment]
> Are you talking about the list not updating as soon as we delete the icon? I think using the collect block instead of forEach would work

Yes, I tried looking into it and it seems the DB doesn't remove those medications, it returns 0 for the method in `MedicationDao`. So we need to fix that first before merging the PR.

File: app/src/main/java/com/waseefakhtar/doseapp/feature/home/usecase/UpdateMedicationUseCase.kt [hunk 1]
```diff
@@ -8,6 +8,9 @@ class UpdateMedicationUseCase @Inject constructor(
     private val repository: MedicationRepository
 ) {
 
+    suspend fun deleteMedication(medication: Medication){
```
Comment (waseefakhtar) [inline app/src/main/java/com/waseefakhtar/doseapp/feature/home/usecase/UpdateMedicationUseCase.kt:11]: We should create a new use case for this method.

**Comment by zaiddkhan** [comment]
Hey @waseefakhtar I've made the changes and pushed the code, and also I've inspected it using the database inspector, the delete functionality is working fine!

File: app/src/main/java/com/waseefakhtar/doseapp/feature/home/HomeScreen.kt [hunk 2]
```diff
@@ -137,73 +179,95 @@ fun DailyMedications(state: HomeState, viewModel: HomeViewModel) {
...
             }
         )
     }
 }
 
+@OptIn(ExperimentalFoundationApi::class)
 @Composable
-fun MedicationCard(medication: Medication, viewModel: HomeViewModel) {
+fun MedicationCard(
+    showDeleteDialog: (medication : Medication) -> Unit,
```
Comment (Wottrich) [inline app/src/main/java/com/waseefakhtar/doseapp/feature/home/HomeScreen.kt:195]: MedicationCard shouldn't known what this HOF will do. This parameter could be called onLongClick without passing medication through the parameter. It increase the component reusability.

Comment (zaiddkhan) [reply app/src/main/java/com/waseefakhtar/doseapp/feature/home/HomeScreen.kt:195]: I've made the changes!

**Comment by waseefakhtar** [comment]
Thank you @zaiddkhan but I'm going to have to close this since this requires more than just deleting it from DB now (we need to take care of alarms scheduling as well) and the delete action does not work as expected.

**Pull request closed without merging**